roles/test_operator/tasks/main.yml

```diff
@@ -143,15 +143,25 @@
     namespace: "{{ cifmw_test_operator_namespace }}"
     kind: Job
     name: "{{ cifmw_test_operator_tempest_name }}"
-    wait: true
-    wait_timeout: "{{ cifmw_test_operator_timeout }}"
-    wait_condition:
-      type: Complete
-      status: true
+  retries: "{{ (cifmw_test_operator_timeout / 10) | round | int }}"
+  delay: 10
+  until: >
+    tempest.resources[0].status.succeeded | default(0) | int >= 1 or
+    tempest.resources[0].status.failed | default(0) | int >= 1
+  ignore_errors: true
+  register: tempest
   when: not cifmw_test_operator_dry_run | bool
 
-- name: Start test-operator-logs-pod
+- name: Check whether tempest timed out
+  ansible.builtin.set_fact:
+    tempest_timed_out: >-
+      {{ tempest.attempts == (cifmw_test_operator_timeout / 10) | round | int }}
   when: not cifmw_test_operator_dry_run | bool
+
+- name: Start test-operator-logs-pod
+  when:
+    - not cifmw_test_operator_dry_run | bool
```

Comment:
would this mean there will be no logs collected in case it hanged/timed-out?

just wild guessing but if its maybe due to access to persistent storage or such, could there be some read-only parallel access, or maybe force terminate the execution on timeout to get access to logs?

Reply:
I wanted to skip these because there are no longs to be collected in case of time out [1]. The logs are moved to the directory where the PV is mounted at the end of the test execution.

However, it is possible that there might be an issue with parallel access to the logs. 

[1] https://github.com/openstack-k8s-operators/tcib/blob/106264ab0a0b18be278d69b0664dad6a4f9be29a/container-images/tcib/base/os/tempest/run_tempest.sh#L265

Reply:
that is not correct

generate logs is post processing the tempest logs.

the tempest logs are stored in a binary format by stester when tempest is run in parrall with the execution fo the these

this https://github.com/openstack-k8s-operators/tcib/blob/106264ab0a0b18be278d69b0664dad6a4f9be29a/container-images/tcib/base/os/tempest/run_tempest.sh#L226C5-L226C66

stestr last --subunit > ${TEMPEST_PATH}testrepository.subunit

is just copying the last log to a new location.

however what we should do doing is making sure that the default .stestr folder
is mind mounted to a log volume so that the logs are directly written to a PVC and we can post process them even when a time out happens.


alternatively we shoudl be passing the time out to the tempest run invocation

timeout ${TEMPEST_TIMEOUT} tempest run ${TEMPEST_ARGS}

hhttps://github.com/openstack-k8s-operators/tcib/blob/106264ab0a0b18be278d69b0664dad6a4f9be29a/container-images/tcib/base/os/tempest/run_tempest.sh#L215


then the job will finish in a roghyly fixed time and it will generate the logs properly then we done need to wait for a fix period of time here in this script.

Reply:
making the run_tempest.sh script enforce the timeout when invoking tempest is obviously the simpler option